src/util/File.h

```diff
@@ -285,6 +286,15 @@ inline void deleteFile(const string& path) {
   if (p != std::string::npos) {
     path_raw.replace(p, 1, "\\ ");
   }
+
+  // Adding escape char for paths with space chars using RE2
+  // < Currently generates undefined reference when building (CMakeLists) >
+  /*
+   string replace = "\\ ";
+   RE2 space = "( )";
+   re2::RE2::Replace(&replace,space,path_raw);
+  */
+
```

Comment:
Yes, `re2` is not header-only, so if you introduce it, you have to link everything that uses `File` against `re2`.
This is done by adding `re2` to the line `target_link_libraries` in the `CMakeLists.txt` files of the targets that fail.
(I would start with `index` and `engine` and see, if this suffices.

Let me know if you have further questions.